docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,155 @@
+    return new static(
+      $container->get('http_client'),
+      $container->get('foia_webform.agency_lookup_service'),
+      $container->get('logger.channel.foia_webform')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function sendSubmissionToComponent(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    $apiUrl = $agencyComponent->get('field_submission_api');
+
+    if ($apiUrl) {
+      $requestData = $this->assembleRequestData($webformSubmission, $agencyComponent);
+    }
+    else {
+      $this->logger
+        ->error('Unable to submit request via the API. Missing API Submission URL for node %component',
+          [
+            '%component' => $agencyComponent->id(),
+            'link' => $agencyComponent->toLink($this->t('Edit Component'), 'edit-form')->toString(),
+          ]
+        );
+    }
+  }
+
+  /**
+   * Gathers the required fields for the API request.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The FOIA form submission form values.
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   *
+   * @return array
+   *   Return the assemble request data in an array.
+   */
+  public function assembleRequestData(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent) {
+    $submissionValues = $this->getSubmissionValues($webformSubmission);
+    $agencyInfo = $this->getAgencyInfo($agencyComponent);
+    $requestValues = array_merge($submissionValues, $agencyInfo);
+
+    return $requestValues;
+  }
+
+  /**
+   * Return the FOIA form submission values as an array.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The Webform submission values.
+   *
+   * @return array
+   *   Returns the submission values as an array.
+   */
+  public function getSubmissionValues(WebformSubmissionInterface $webformSubmission) {
+    return $webformSubmission->getData();
+  }
+
+  /**
+   * Parse the Agency information from the Agency Component.
+   *
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   */
+  public function getAgencyInfo(NodeInterface $agencyComponent) {
+    return [
+      'agency' => $this->agencyLookUpService->getAgencyFromComponent($agencyComponent),
+      'agency_component_name' => $agencyComponent->label(),
+    ];
+  }
+
+  /**
+   * Get the metadata for the file attachments.
+   *
+   * @param array $files
+   *   An array containing the file IDs for the file attachments.
+   */
+  public function getAttachmentData(array $files) {
+    if (!empty($files)) {
+      $fileData = [];
+      foreach ($files as $fid) {
+        $currentFile = File::load($fid);
+        print_r($currentFile);
+        $base64 = base64_encode(file_get_contents($currentFile->getFileUri()));
```

Comment:
I tend to try and avoid alluding to the type or data specifics of the variable being stored. In this case we're holding file contents, and that would be the case even if we leveraged some sort of other encoding. So maybe this variable can be renamed `$fileContents`. There's a great book titled 'Clean Code' that talks about this. I see it as a form of 'Hungarian Notation'.

"...nowadays HN and other forms of type encoding are simply impediments. They make it harder to change the name or type of a variable, function, member or class. They make it harder to read the code. And they create the possibility that the encoding system will mislead the reader."